### DISCUSSION TITLE
*trying* to access the value without tracking, use `.get_untracked()` or `.with_untracked()` instead warning in use_params_map()

### DISCUSSION THREAD

**Discussion by silicology1** [leptos-rs/leptos#2451] [Q&A]
```rust
#[component]
pub fn SchellingGame() -> impl IntoView {
    let params = use_params_map();
    let profile_user_account = move || {
        params.with(|params| { // error line number
            params
                .get("profile_user_account")
                .cloned()
                .unwrap_or_default()
        })
    };

    view! {
        <div>
        <SchellingGameComponent profile_user_account=profile_user_account()/>
        </div>
    }
}

#[component]
pub fn SchellingGameComponent(profile_user_account: String) -> impl IntoView {
    let period_fn: ReadSignal<Option<Period>> = get_period_fn(profile_user_account);
view {
  <div>
{move || format!("{:?}", period())}
  </div>
}
}
```
profile_user_account also don't have get_untracked() function, how to solve the error?

At src/components/schelling_game/profile_validation/game/schelling_game.rs:16:16, you access a signal or memo (defined at /home/amiya/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_router-0.6.6/src/components/route.rs:280:22) outside a reactive tracking context. This might mean your app is not responding to changes in signal values in the way you expect.

Here’s how to fix it:

1. If this is inside a `view!` macro, make sure you are passing a function, not a value.
  ❌ NO  <p>{x.get() * 2}</p>
  ✅ YES <p>{move || x.get() * 2}</p>

2. If it’s in the body of a component, try wrapping this access in a closure: 
  ❌ NO  let y = x.get() * 2
  ✅ YES let y = move || x.get() * 2.

3. If you’re *trying* to access the value without tracking, use `.get_untracked()` or `.with_untracked()` instead.

**Comment by gbj** [comment]
You can always wrap anything in [`untrack`](https://docs.rs/leptos/latest/leptos/fn.untrack.html).
